### PR TITLE
Add _plugins/_notifications/channels API spec

### PR DESCRIPTION
**Pull request by spapadop**
### Description
Adds the one missing API of the notifications namespace

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-api-specification/issues/229

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/namespaces/notifications.yaml [hunk 1]
```diff
@@ -4,6 +4,17 @@ info:
...
   version: 1.0.0
 paths:
+  /_plugins/_notifications/channels:
+    get:
+      operationId: notifications.list_channels.0
+      x-operation-group: notifications.list_channels
+      x-version-added: '2.0'
+      description: List created notification channels.
+      externalDocs:
+        url: https://opensearch.org/docs/latest/observing-your-data/notifications/api/#list-all-notification-configurations
```
Comment (spapadop) [inline spec/namespaces/notifications.yaml:14]: there is no good link currently on the docs, I'll try to raise a request there to add one

Comment (spapadop) [reply spec/namespaces/notifications.yaml:14]: https://github.com/opensearch-project/documentation-website/pull/6982

Comment (nhtruong) [reply spec/namespaces/notifications.yaml:14]: Thank you. Let's add a `TODO` comment saying that the URL needs to be updated once we have a proper URL.

File: spec/namespaces/notifications.yaml [hunk 2]
```diff
@@ -225,6 +236,12 @@ components:
         application/json:
           schema:
             $ref: '../schemas/notifications._common.yaml#/components/schemas/GetConfigsResponse'
+    notifications.list_channels@200:
+      description: ''
+      content:
+        application/json:
+          schema:
+            $ref: '../schemas/notifications._common.yaml#/components/schemas/ListChannelsResponse'
```
Comment (nhtruong) [inline spec/namespaces/notifications.yaml:244]: is `ListChannelsResponse` going to be reused reused else where? If not we should just declare the schema here to reduce reference hell. The other endpoints that have this kind of references came from the legacy Smithy->OpenAPI conversion. We will need to update them in a future PR.

Comment (spapadop) [reply spec/namespaces/notifications.yaml]: Good point! done.

File: spec/namespaces/notifications.yaml (hunk 1 above)
Comment (spapadop) [reply spec/namespaces/notifications.yaml:15]: I hope it will be merged soon, but added the TODO for now.

**Review by nhtruong — APPROVED** [review]

**Pull request merged**